### PR TITLE
added auth functionality

### PR DESCRIPTION
**Pull request by aishwarya-bm**
added signup/login features with validation for email & mobile
used context,reducer to handle auth

preview:
https://user-images.githubusercontent.com/66695068/160245845-96118010-71f7-4d52-8935-3683c3f9a9c6.mp4

### CODE REVIEW

File: src/components/signup/Signin.jsx [hunk 1]
```diff
@@ -1,30 +1,65 @@
...
-    // return userDispatch({ type: "USER_FORM", payload: event.target });
+    setLoginForm(() => ({
+      ...loginForm,
+      [event.target.name]: event.target.value,
+    }));
   };
 
   const toggleShowPassword = e => {
     e.preventDefault();
-    console.log("in toggle pwd");
```
Comment (shubhamkumar648) [inline src/components/signup/Signin.jsx:16]: remove console.log when you push in production

File: src/components/signup/Signup.jsx [hunk 2]
```diff
@@ -1,24 +1,87 @@
...
 
+  const createUser = async signupForm => {
+    const userErrors = validateUser(signupForm.mobile, signupForm.email);
+    setUserErr(userErrors);
+    if (userErrors["phone"] || userErrors["email"]) {
+      return;
+    }
+    try {
+      const response = await axios.post("api/auth/signup", signupForm);
+      if (response.status === 201) {
```
Comment (shubhamkumar648) [inline src/components/signup/Signup.jsx:43]: you can also use ternary operator here

**Review by shubhamkumar648 — APPROVED** [review]
consider comment as suggestion .the remaining structure of code  is good

File: src/components/signup/Signin.jsx (hunk 1 above)
Comment (aishwarya-bm) [reply src/components/signup/Signin.jsx:16]: this is already a removed code line

File: src/components/signup/Signup.jsx (hunk 2 above)
Comment (aishwarya-bm) [reply src/components/signup/Signup.jsx:43]: removed the empty else part

File: src/components/signup/Signin.jsx [hunk 3]
```diff
@@ -1,30 +1,65 @@
...
+    setShowAlert(true);
+
+    setTimeout(() => {
+      setShowAlert(false);
+    }, 5000);
+  };
+
+  const loginUser = async loginForm => {
+    try {
+      let response = await axios.post("api/auth/login", loginForm);
```
Comment (SimranjitKaurMaan) [inline src/components/signup/Signin.jsx]: You can use `const` instead of `let`

**Review by raman1999 — APPROVED** [review]
code looks clean to me, good component breakdown done

File: src/components/signup/Signin.jsx (hunk 3 above)
Comment (aishwarya-bm) [reply src/components/signup/Signin.jsx]: made the change

**Pull request merged**